packages/react-components/react-carousel-preview/library/src/components/Carousel/useCarousel.ts

```diff
@@ -62,6 +71,42 @@ export function useCarousel_unstable(props: CarouselProps, ref: React.Ref<HTMLDi
 
   const mergedRefs = useMergedRefs(ref, containerRef);
 
+  // Announce carousel updates
+  const announcementTextRef = React.useRef<string>('');
+  const totalNavLength = React.useRef<number>(0);
+  const navGroupRef = React.useRef<number[][]>([]);
+
+  const { announce } = useAriaLiveAnnouncer_unstable(props);
+
+  const updateAnnouncement = useEventCallback(() => {
+    if (totalNavLength.current <= 0 || !announcement) {
+      // Ignore announcements until slides discovered
+      return;
+    }
+
+    const announcementText = announcement(activeIndex, totalNavLength.current, navGroupRef.current);
+
+    if (announcementTextRef.current === '') {
```

Comment:
I believe that this will cause bugs in Strict Mode, https://react.dev/reference/react/StrictMode#fixing-bugs-found-by-double-rendering-in-development